airflow/executors/mesos_executor.py

```diff
@@ -0,0 +1,380 @@
+    MesosExecutor allows distributing the execution of task
+    instances to multiple mesos workers.
+
+    Apache Mesos is a distributed systems kernel which abstracts
+    CPU, memory, storage, and other compute resources away from
+    machines (physical or virtual), enabling fault-tolerant and
+    elastic distributed systems to easily be built and run effectively.
+    See http://mesos.apache.org/
+    """
+
+    class MesosFramework(threading.Thread):
+        """MesosFramework class to start the threading"""
+
+        def __init__(self, client):
+            super().__init__(target=self)
+            self.client = client
+            self.stop = False
+
+        def run(self):
+            try:
+                self.client.register()
+            except KeyboardInterrupt:
+                print("Stop requested by user, stopping framework....")
+
+    def __init__(self):
+        super().__init__()
+        self.commands_to_run = []
+        self.task_queue = Queue()
+        self.result_queue = Queue()
+        self.driver = None
+        self.client = None
+        self.mesos_framework = None
+
+    @provide_session
+    def start(self, session=None):
+        """Setup and start routine to connect with the mesos master"""
+        master = conf.get('mesos', 'MASTER')
+
+        framework_name = get_framework_name()
+        framework_id = None
+        framework_role = conf.get('mesos', 'FRAMEWORK_ROLE', fallback="marathon")
+
+        task_cpu = conf.getint('mesos', 'TASK_CPU', fallback=1)
+        task_memory = conf.getint('mesos', 'TASK_MEMORY', fallback=256)
+
+        if conf.getboolean('mesos', 'CHECKPOINT'):
+            framework_checkpoint = True
+
+            if conf.get('mesos', 'FAILOVER_TIMEOUT'):
+                # Import here to work around a circular import error
+                from airflow.models import Connection
+
+                # Query the database to get the ID of the Mesos Framework, if available.
+                conn_id = FRAMEWORK_CONNID_PREFIX + framework_name
+                connection = session.query(Connection).filter_by(conn_id=conn_id).first()
+                if connection is not None:
+                    # Set the Framework ID to let the scheduler reconnect
+                    # with running tasks.
+                    framework_id = connection.extra
+
+                # Set Timeout in the case of a mesos master leader change
+                framework_failover_timeout = conf.getint('mesos', 'FAILOVER_TIMEOUT')
+
+        else:
+            framework_checkpoint = False
+
+        self.log.info(
+            'MesosFramework master : %s, name : %s, cpu : %d, mem : %d, checkpoint : %s, id : %s',
+            master,
+            framework_name,
+            task_cpu,
+            task_memory,
+            framework_checkpoint,
+            framework_id,
+        )
+
+        master_urls = "https://" + master
+
+        self.client = MesosClient(
+            mesos_urls=master_urls.split(','),
```

Comment:
This seems odd -- won't this only prefix the _first_ connection with `https://` -- do we need to split first then make sure every url has the scheme?

Reply:
It looks odd, yes. :-)  But the `mesos_urls` parameter of the MesosClient library have to be a list. And since these parameter will not have more then one item, the connection should ever `https`. What I'm thinking now, if I take out the protocol and let the user choose. Means, the user define the protocol into the MASTER variable. Because, maybe someone want to run Mesos without SSL. What do u think?